src/Analyser/NodeScopeResolver.php

```diff
@@ -1410,10 +1411,16 @@ private function processStmtNode(
 				}
 			}
 
+			if ($alwaysIterates) {
+				$isAlwaysTerminating = count($finalScopeResult->getExitPointsByType(Break_::class)) === 0;
+			} else {
+				$isAlwaysTerminating = false; // $finalScopeResult->isAlwaysTerminating() && $isAlwaysIterable
+			}
+
 			return new StatementResult(
 				$finalScope,
 				$finalScopeResult->hasYield() || $hasYield,
-				false/* $finalScopeResult->isAlwaysTerminating() && $isAlwaysIterable*/,
+				$isAlwaysTerminating,
```

Comment:
I feel like I'm missing `$finalScopeResult->isAlwaysTerminating()` here. It's related to the failing test we talk above.

Reply:
Also please update the E2E test so it's not failing.

Reply:
adapted I think. not sure if I did it the right way, but I manually checked-out PHP-Parser and manually run PHPStan like in the test and let it update the baseline there.